x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/IndicesPermission.java

```diff
@@ -64,6 +64,38 @@ public IndicesPermission(Group... groups) {
     }
 
     static Predicate<String> indexMatcher(List<String> indices) {
+        Set<String> exactMatch = new HashSet<>();
+        List<String> nonExactMatch = new ArrayList<>();
+        for (String indexPattern : indices) {
+            if (indexPattern.startsWith("/")) {
```

Comment:
The existing `Automatons` code throws an exception if a pattern starts with `/` but doesn't end with `/`.
So we could simplify all this checking down to a single if/else and still be compatible with existing behaviour.
```
if (indexPattern.startsWith("/") || indexPattern.contains("*") || indexPattern.contains("?")) {
    nonExactMatch.add(indexPattern);
} else {
    exactMatch.add(indexPattern);
}
```

I don't care strongly, but shorter and simpler seems nicer.

Reply:
++ thanks for noticing that. I misread the if statement in Automatons